### ISSUE TITLE
Flagsmith Hangs Indefinitely

### ISSUE THREAD

**Issue by fhu-klwines** [Flagsmith/flagsmith-dotnet-client#105]
Hello,

Our team is recently exploring the option of using Flagsmith on our application running ***.net framework 4.7.3***
During our implementation, we discovered two cases where it leads to indefinite hanging when trying to get flags from flagsmith.

### Case 1 : Initializing Client with CacheConfig

In our testing, we have found that setting `CacheConfig` results in indefinite hanging issue when `FlagsmithClient.GetEnvironmentFlags` is called

```c#
var cacheConfig = new CacheConfig(true)
            {
                DurationInMinutes = 1
            };
var flagsmithConfiguration = new FlagsmithConfiguration
            {
                EnvironmentKey = flagsmithEnvKey,
                CacheConfig = cacheConfig
            };
FlagsmithClient = new FlagsmithClient(flagsmithConfiguration);
```

### Case 2 : Getting Environment Flags Synchronously

Based on the [docs](https://docs.flagsmith.com/clients/server-side?language=dotnet), we were not able to get the environment flags synchronously and observed that the call also hangs indefinitely. Please note that at this point, we have disabled `CacheConfig` setting to avoid issue from case 1.

The issue only happens when we tried to hit it synchronously. If we were to await the call, the flags do come back swiftly.

I.E.
`flagsmithClient.GetEnvironmentFlags().Result;` -> hangs
`await flagsmithClient.GetEnvironmentFlags();` -> works just fine

Please keep in mind that we are using ***.net framework 4.7.3*** and do not have the capability to upgrade to later versions.

Any help is greatly appreciated. Please let me know if more information is needed.
Thank you!

**Comment by matthewelwell** [comment]
Thanks @fhu-klwines , we'll take a look into this and get back to you as soon as we can.

**Comment by novakzaballa** [comment]
@fhu-klwines just to keep you in the loop, I will start looking at this issue now. I'll keep this thread up to date with any new information.

**Comment by DavidPerish** [comment]
@fhu-klwines we're seeing a similar issue when invoking GetFlag(...).Result from an MVC 4.7 razor view.  I'm curious if you're trying to access this via razor as well, or if you're seeing this in controllers or elsewhere?

**Comment by fhu-klwines** [comment]
@DavidPerish Yes, we are observing this behavior in razor and in controllers

**Comment by DavidPerish** [comment]
> @DavidPerish Yes, we are observing this behavior in razor and in controllers

I see, thanks.  Are you using the analytics features, and does this still happen if you set EnableAnalytics to false on the FlagsmithClient constructor or in FlagsmithConfiguration?

**Comment by fhu-klwines** [comment]
We are using default setting atm, but even if I set analytics to false, it still hangs

**Comment by novakzaballa** [comment]
A [PR](https://github.com/Flagsmith/flagsmith-dotnet-client/pull/107) is now being reviewed to solve this issue.

**Comment by novakzaballa** [comment]
Hi @fhu-klwines, this issue was resolved in the new version 5.3.1.
